mllib/src/main/scala/org/apache/spark/mllib/tree/impurity/Entropy.scala

```diff
@@ -83,23 +83,29 @@ object Entropy extends Impurity {
  * @param numClasses  Number of classes for label.
  */
 private[spark] class EntropyAggregator(numClasses: Int)
-  extends ImpurityAggregator(numClasses) with Serializable {
+  extends ImpurityAggregator(numClasses + 1) with Serializable {
```

Comment:
Quick q: why + 1 here?

Reply:
that was my question here too: https://github.com/apache/spark/pull/16722/files#r99274459
and after reading the code I came up with the answer:
`
I guess it is because the number of "stats" increases by one since we are adding the numSamples in addition to numSamples * weight, if I understand correctly
`